evap/staff/views.py

```diff
@@ -157,55 +142,71 @@ def semester_course_operation(request, semester_id):
             helper_semester_course_operation_revert(request, courses)
         elif target_state == 'prepared':
             helper_semester_course_operation_prepare(request, courses, template)
-        elif target_state == 'approved':
-            helper_semester_course_operation_approve(request, courses)
         elif target_state == 'in_evaluation':
             helper_semester_course_operation_start(request, courses, template)
-        elif target_state == 'published':
-            helper_semester_course_operation_publish(request, courses, template)
         elif target_state == 'reviewed':
             helper_semester_course_operation_unpublish(request, courses)
+        elif target_state == 'published':
+            helper_semester_course_operation_publish(request, courses, template)
+
 
         return custom_redirect('staff:semester_view', semester_id)
 
     # If necessary, filter courses and set email template for possible editing
     email_template = None
     if courses:
-        if target_state == 'prepared':
-            email_template = EmailTemplate.objects.get(name=EmailTemplate.EDITOR_REVIEW_NOTICE)
-
-        elif target_state == 'approved':
-            # remove courses without questionnaires on general contribution, warn about courses with missing questionnaires
-            courses_with_enough_questionnaires = [course for course in courses if course.general_contribution_has_questionnaires]
-            courses_with_missing_questionnaires = [course for course in courses_with_enough_questionnaires if not course.all_contributions_have_questionnaires]
+        if target_state == 'new':
```

Comment:
previously there wasn't any code for this? why?

Reply:
we now have buttons for all course operations on one page. not all operations are valid for courses in all states. we needed less checks before, because the available operations were per state.